File: php/features.php
<?php
declare(strict_types=1);

// PublicApisDatabase SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PublicApisDatabaseFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PublicApisDatabaseBaseFeature();
            case "test":
                return new PublicApisDatabaseTestFeature();
            default:
                return new PublicApisDatabaseBaseFeature();
        }
    }
}

File: py/features.py
# PublicApisDatabase SDK feature factory

from feature.base_feature import PublicApisDatabaseBaseFeature
from feature.test_feature import PublicApisDatabaseTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PublicApisDatabaseBaseFeature(),
        "test": lambda: PublicApisDatabaseTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
